src/win/thread.c

```diff
@@ -191,7 +191,13 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 
 
 uv_thread_t uv_thread_self(void) {
-  return (uv_thread_t) uv_key_get(&uv__current_thread_key);
+  void* self;
+  uv_once(&uv__current_thread_init_guard, uv__init_current_thread_key);
+  self = uv_key_get(&uv__current_thread_key);
+  if (self == NULL)
+      /* This thread wasn't created with uv_thread_create. */
+      self = GetCurrentThread();
```

Comment:
Note that GetCurrentThread() always returns a pseudo-thread handle which is only valid in the context of the current thread, and if I remember correctly, the same for all threads. So returning NULL (which is the default value of the TLS slot) is just as good for the purpose of comparing uv_thread_t values returned by uv_thread_self().

https://msdn.microsoft.com/en-us/library/windows/desktop/ms683182(v=vs.85).aspx

In summary, I would recommend no change here, and move the uv__init_current_thread_key initialization to uv_init()